### ISSUE TITLE
helper:assert-throws-error, xdmp:apply, varargs

### ISSUE THREAD

**Issue by kefo** [marklogic/roxy#503]
I have a function I want to test.  It takes 6 parameters.  If I call `helper:assert-throws-error`, I get a “Too few args” error message.  The error is actually occurring at line [295](https://github.com/marklogic/roxy/blob/master/src/test/test-helper.xqy#L295).

I think this might be a bug because xdmp:apply seems to require the $params be itemized.  Instead of this

```xquery
	let $params := (‘1’, ‘2’, ‘3’)
	return xdmp:apply( $func, $params )
```
You seem to have to do this:

```xquery
	let $params := (‘1’, ‘2’, ‘3’)
	return xdmp:apply($func, $params[1], $params[2], $params[3])
```

(Side note:  Some googling turned up this [2011 post](https://www.mail-archive.com/general@developer.marklogic.com/msg07547.html) discussing the ‘varargs’ aspect of `xdmp:apply`.)

I’ve actually implemented a fix locally and it seems to work.  Of course, it has all ugliness mentioned in the above post.  Happy to submit a pull request if, in fact, this is an issue.

If it isn’t an issue, please advise how one uses `helper:assert-throws-error` with multiple vars.

**Comment by grtjn** [comment]
I think you found the best solution for the moment, though I wonder if you need to distinguish between all numbers of args like in the 2011 post. That link also proposes to use map:map to pass through arguments, which solves passing through sequences as argument. Or perhaps json:array would be practical here. Definitely a bug..

**Comment by grtjn** [comment]
Fixed in dev..

**Comment by tapuzzo-fsi** [comment]
This is not actually working from what I can tell.  There are 2 problems that i see, one is that the way the cascade of function calls are built, it will not work with <6 parameters.  What ends up happening is that all sample functions get called with 6 parameters, with the balance being the empty sequence.

Once that was corrected, I noticed a different problem/limitation in `json:to-array`.  There is some sequence collapse processing happening somewhere in the call stack that shouldn't be.  This problem can be seen with this: `json:to-array(json:to-array(1))` which should yield `[ [ 1 ] ]` but actually yields `[ 1 ]`.  Luckily there is a way to workaround this by doing `json:to-array((json:to-array(1), json:to-array('xtra')), 1)`

See attached patch for a fix: 
[refactored-json-array-patch.txt](https://github.com/marklogic/roxy/files/292413/refactored-json-array-patch.txt)

I can make a pull-request if that is helpful, but didn't seem necessary for such a small diff.
